### PR TITLE
add get topics endpoint and general error handling

### PR DESCRIPTION
**Pull request by bryantomson**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("path not found");
+      });
+  });
+  test("POST:406 responds with 406 when POST request receieved for path where method is not allowed", () => {
+    return request(app)
+      .post("/api/topics")
+      .expect(406)
```
Comment (two-triangles) [inline __tests__/app.test.js]: All of these 406 tests can be removed as we will be introducing these methods later on which means that these tests will fail.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const { getTopics } = require("./controller");
+const {  handleCustomErrors, handle505, handleNotFound } = require("./errors");
+
+const express = require("express");
+const app = express();
+app.use(express.json());
```
Comment (two-triangles) [inline app.js]: this middleware can be removed as we are not doing POST or PATCH requests yet

File: errors.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+
+exports.handleCustomErrors = (err, req, res, next) => {
+  next(err);
+};
+
+exports.handleNotFound = (err, req, res, next) => {
+  if (err.status) {
+    res.status(err.status).send({ msg: err.msg });
+  }
+  next();
```
Comment (two-triangles) [inline errors.js]: next should be invoked with the error

File: errors.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+};
+
+exports.handle505 = (err, req, res, next) => {
+     res.status(505).send({ msg: "internal server error" });
+    next();
```
Comment (two-triangles) [inline errors.js]: This error handler won't need next as it should be the last error handler in the middleware chain

**Review by johnhutcheon — APPROVED** [review]
Thanks for making the changes Tom! Just make sure you change the 505 to a 500, this is the error code for an internal server error 👍

**Pull request merged**